// astro/dist/cli/info/infra/cli-clipboard.js
async function getExecInputForPlatform({
  platform,
  commandExecutor
}) {
  if (platform === "darwin") {
    return ["pbcopy"];
  }
  if (platform === "win32") {
    return ["clip"];
  }
  const unixCommands = [
    ["xclip", ["-selection", "clipboard"]],
    ["wl-copy", []]
  ];
  for (const [unixCommand, unixArgs] of unixCommands) {
    try {
      const { stdout } = await commandExecutor.execute("which", [unixCommand]);
      if (stdout.trim()) {
        return [unixCommand, unixArgs];
      }
    } catch {
      continue;
    }
  }
  return null;
}
function createCliClipboard({
  operatingSystemProvider,
  commandExecutor,
  logger,
  prompt
}) {
  return {
    async copy(text) {
      text = text.trim();
      const platform = operatingSystemProvider.name;
      const input = await getExecInputForPlatform({ platform, commandExecutor });
      if (!input) {
        logger.warn("SKIP_FORMAT", "Clipboard command not found!");
        logger.info("SKIP_FORMAT", "Please manually copy the text above.");
        return;
      }
      if (!await prompt.confirm({
        message: "Copy to clipboard?",
        defaultValue: true
      })) {
        return;
      }
      try {
        const [command, args] = input;
        await commandExecutor.execute(command, args, {
          input: text,
          stdio: ["pipe", "ignore", "ignore"]
        });
        logger.info("SKIP_FORMAT", "Copied to clipboard!");
      } catch {
        logger.error(
          "SKIP_FORMAT",
          "Sorry, something went wrong! Please copy the text above manually."
        );
      }
    }
  };
}
export {
  createCliClipboard
};
